subtrack.MAUI/Services/SubscriptionsCalculator.cs

```diff
@@ -95,4 +91,31 @@ public DateTime GetNextPaymentDate(Subscription subscription)
         var nextPaymentDate = GetNextPaymentDate(subscription);
         return (nextPaymentDate <= _dateProvider.Today, nextPaymentDate);
     }
+
+    private static bool PaidThisMonth(Subscription subscription, DateTime monthDate) => subscription.LastPayment.Month == monthDate.Month && subscription.LastPayment.Year == monthDate.Year;
+
+    private static bool PaidInPast(Subscription subscription, DateTime monthDate) => (subscription.LastPayment.Month < monthDate.Month && subscription.LastPayment.Year == monthDate.Year) || subscription.LastPayment.Year < monthDate.Year;
```

Comment:
This doesn't work for a weekly subscription that has last payment this month. If im not mistaken we need access to the next payment in order to know if it should be paid this month

Reply:
It won't fetch any subscriptions that have last payment this month. This is based on the existing behaviour.

```
 var subscriptionsByMonth = subscriptions
                    .Where(s => !paidThisMonth(s) && !paymentStartsInTheFuture(s))
                    .ToList();
 ```

Reply:
Yes because we only had monthly subs before so if a sub had been paid this month wed know that it shouldnt be included. We need to add a special case for weekly subs